autodiff/reverse/var/var.hpp

```diff
@@ -254,6 +255,9 @@ struct Expr
     /// Destructor (to avoid warning)
     virtual ~Expr() {}
 
+    virtual void bind_value(T* /* grad */) {}
```

Comment:
Please add a short (Doxygen ///) comment here for these `bind_*` functions. I like your approach.